### PR TITLE
Adding the artist data on mouse move event message

### PR DESCRIPTION
**Pull request by fariza**
In the tools we didn't add the artist data to the message on mouse move event.

Also added a small check, for `get_visible` artist list, this is because if the artist is not visible we shouldn't show its data

### CODE REVIEW

File: lib/matplotlib/backend_tools.py [hunk 1]
```diff
@@ -335,6 +335,16 @@ def send_message(self, event):
             except (ValueError, OverflowError):
                 pass
             else:
+                artists = [a for a in event.inaxes.mouseover_set
+                           if a.contains(event) and a.get_visible()]
+
+                if artists:
+                    a = max(enumerate(artists), key=lambda x: x[1].zorder)[1]
```
Comment (Kojoley) [inline lib/matplotlib/backend_tools.py]: Is not `a = max(artists, key=lambda x: x.zorder)` is the same as this?

Comment (fariza) [inline lib/matplotlib/backend_tools.py]: My guess is that it is but to be honest, I didn't check, I just copy/paste from `backend_bases.py`

**Comment by fariza** [comment]
The failures are not related

File: lib/matplotlib/backend_tools.py (hunk 1 above)
Comment (fariza) [inline lib/matplotlib/backend_tools.py]: @tacaswell as you are the original author of that line, do you see a problem with @Kojoley suggestion?

Comment (tacaswell) [inline lib/matplotlib/backend_tools.py]: I looked at this earlier today and have a memory of doing it this way intentionally, but no longer remember the _why_.

The discussion at https://github.com/matplotlib/matplotlib/pull/4813/files is not enlightening and I can not even claim I just carried through some older code.  

I think making that change is fine.

**Comment by fariza** [comment]
@Kojoley I simplified as suggested. It works fine on my side

**Comment by tacaswell** [comment]
backported to v2.x as f52af63

**Pull request merged**